src/Iterators.jl

```diff
@@ -183,7 +188,7 @@ function start(it::Product)
             return js, nothing
         end
     end
-    vs = Array(Any, n)
+    vs = Array{Any}(n)
```

Comment:
it's good to do `Vector` for this so the compiler can do less work with determining the number of dimensions